### PR TITLE
INFINITY-[1532,1533] First pass at releasing testing/+tools/ .whl's

### PR DESCRIPTION
**Pull request by nickbp**
Adds tooling which builds a `.whl` file for each of `testing/` and `tools/`, and updates the release script to upload these `.whl` files to S3.

The provided `tools/pip/build.sh` includes example steps for both executing tools and for importing testing libs. In theory external repos can just point their builds to these `.whl`s and avoid the need to manually copy `testing/` and `tools/`, but for now this is really just setting up a source of truth for use in future tooling.

### CODE REVIEW

File: tools/pip/setup.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
```
Comment (benclarkwood) [inline tools/pip/setup.py]: Are these terrifying looking python files just copied boilerplate from somewhere else?

Comment (nickbp) [reply tools/pip/setup.py]: This is partially copied from the prior CLI python wrapper (used for 1.7 compat), which was also a `.whl`.

File: release.sh [hunk 2]
```diff
@@ -112,8 +114,17 @@ fi
...
+
 aws s3 cp --acl public-read sdk/bootstrap/bootstrap.zip $S3_DIR/bootstrap.zip 1>&2
+
 aws s3 cp --acl public-read sdk/executor/build/distributions/executor.zip $S3_DIR/executor.zip 1>&2
-aws s3 cp --acl public-read sdk/cli/dcos-service-cli-linux $S3_DIR/dcos-service-cli-linux 1>&2
-aws s3 cp --acl public-read sdk/cli/dcos-service-cli-darwin $S3_DIR/dcos-service-cli-darwin 1>&2
-aws s3 cp --acl public-read sdk/cli/dcos-service-cli.exe $S3_DIR/dcos-service-cli.exe 1>&2
+
+for cli_exe in $(ls sdk/cli/dcos-service-cli*); do
+    aws s3 cp --acl public-read $cli_exe $S3_DIR/$(basename $cli_exe) 1>&2
```
Comment (elezar) [inline release.sh]: Does `aws s3 cp` not support copying multiple files? An alternative may be `aws s3 sync` which does.

File: release.sh [hunk 3]
```diff
@@ -112,8 +114,17 @@ fi
...
+done
+
+# note: pip is picky about version formatting, so the outputted filenames may not exactly match $SDK_VERSION due to some mangling.
+# additionally, pip will refuse to handle files without the extra info in the filename, so we leave it in.
+for whl_file in $(ls tools/pip/*.whl); do
```
Comment (elezar) [inline release.sh]: Same comment here regarding `aws s3 cp` vs `aws s3 sync`.

File: sdk/cli/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Default CLI
+
+This directory contains the default CLI build, used by the majority of services who don't add custom commands to their service CLIs. The default CLI builds are uploaded with every SDK release as of early 2018.
```
Comment (elezar) [inline sdk/cli/README.md]: Could we reference a SHA here?

File: release.sh (hunk 3 above)
Comment (nickbp) [reply release.sh]: Done

File: sdk/cli/README.md (hunk 4 above)
Comment (nickbp) [reply sdk/cli/README.md]: Updated the package tooling to support template params like `{{sha256:filename@http://.../SHA256SUMS}}`. Now packages consuming default CLIs will be able to just specify the manifest URL for the SDK release they're using and the sha256sum will be fetched from there automatically.

**Pull request merged**